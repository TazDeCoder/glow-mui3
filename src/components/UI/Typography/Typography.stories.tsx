import type { ComponentMeta, ComponentStory } from '@storybook/react';

import { Typography } from './Typography';
import { Box } from '../../Utils';

export default {
  title: 'Components/UI/Typography',
  component: Typography,
  argTypes: {
    onClick: { action: 'clicked' },
  },
} as ComponentMeta<typeof Typography>;

export const Default: ComponentStory<typeof Typography> = (args) => (
  <Typography {...args}>Sample Text</Typography>
);

export const Variants: ComponentStory<typeof Typography> = (args) => (
  <Box sx={{ display: 'flex', flexDirection: 'column', rowGap: 2 }}>
    <Typography {...args} variant="headline" size="sm">
      Headline Small
    </Typography>
    <Typography {...args} variant="body" size="md">
      Body Medium
    </Typography>
    <Typography {...args} variant="label" size="lg">
      Label Large
    </Typography>
  </Box>
);
