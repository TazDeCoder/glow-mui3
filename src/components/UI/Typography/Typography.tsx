/** @jsxRuntime classic */
/** @jsx jsx */

import { forwardRef } from 'react';
import { jsx } from '@theme-ui/core';

import { Box, BoxProps } from '../../Utils';
import type { ForwardRef, TextVariant, TextSize } from '../../types';

export interface TypographyProps extends BoxProps {
  /** Text variant of the component
   * @default 'body'
   */
  variant?: TextVariant;
  /** Text size of the component
   * @default 'md'
   */
  size?: TextSize;
}

export const Typography: ForwardRef<HTMLParagraphElement, TypographyProps> =
  forwardRef(
    (
      {
        as,
        variant = 'body',
        size = 'md',
        color = 'primary.main',
        sx,
        ...props
      }: TypographyProps,
      ref,
    ) => {
      let Component: React.ElementType = 'p';

      if (!as) {
        if (variant === 'headline') {
          if (size === 'sm') {
            Component = 'h3';
          }
        }
        if (variant === 'label') {
          Component = 'span';
        }
      }

      return (
        <Box
          ref={ref}
          as={Component}
          color={color}
          variant={`${variant}.${size}`}
          sx={sx}
          {...props}
        />
      );
    },
  );
