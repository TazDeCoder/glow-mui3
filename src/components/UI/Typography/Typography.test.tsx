import { render } from '@testing-library/react';

import { Typography } from './Typography';

describe('<Typography />', () => {
  describe('Snapshots', () => {
    it('renders correctly', () => {
      const { container } = render(<Typography>Foo</Typography>);
      expect(container).toMatchSnapshot();
    });
  });
});
