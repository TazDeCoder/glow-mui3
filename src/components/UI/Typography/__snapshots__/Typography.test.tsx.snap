// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Typography /> Snapshots renders correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: primary.main;
}

<div>
  <p
    class="emotion-0"
  >
    Foo
  </p>
</div>
`;
